website/docs/r/token.html.markdown

```diff
@@ -0,0 +1,65 @@
+---
+layout: "rancher2"
+page_title: "Rancher2: rancher2_token"
+sidebar_current: "docs-rancher2-resource-token"
+description: |-
+  Provides a Rancher v2 Token resource. This can be used to create tokens for Rancher v2 provider user and retrieve their information.
+---
+
+# rancher2\_token
+
+Provides a Rancher v2 Token resource. This can be used to create Tokens for Rancher v2 provider user and retrieve their information.
+
+There are 2 kind of tokens:
+- no scoped: valid for global system.
+- scoped: valid for just a specific cluster (`cluster_id` should be provided).
+
+Tokens can't be updated once created. Any diff in token data will recreate the token. If any token expire, Rancher2 provider will generate a diff to regenerate it.
+
+## Example Usage
+
+```hcl
+# Create a new rancher2 Token
+resource "rancher2_token" "foo" {
+  description = "foo token"
+  ttl = 1200
+}
+# Create a new rancher2 Token scoped
+resource "rancher2_token" "foo" {
+  cluster_id = "<cluster-id>"
+  description = "foo token"
+  ttl = 1200
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `cluster_id` - (Optional/ForceNew) Cluster ID for scoped token (string)
+* `description` - (Optional/ForceNew) Token description (string)
+* `enabled` - (Optional/ForceNew) Token is enabled. This argument doesn't need to be provided, it just control and would generate a terraform diff to renew the token in case it's disabled. If i Default `true` (bool)
+* `expired` - (Optional/ForceNew) Token is expired. This argument doesn't need to be provided, it just control and would generate a terraform diff to renew the token in case it's expired. Default `false` (bool)
```

Comment:
With this understanding, should we change the option to `renew_expired`? I think that describes what it does better. Would we really not want to renew expired tokens? I feel like you would want to always renew if you are generating it programmatically.

Reply:
For sure `renew_expired` described what it does better. The problem is that `enabled` and `expired` attributes are still needed to make that terraform notice the change once token is disabled or expired on Rancher side. 

Normally, you would always want to renew tf generated tokens and this is the default resource behaviour and the main reason to add `enabled` and `expired` attributes. If the token is expired or disabled, tf notice the diff and renew the token if you apply tf plan. But eventually for temporary token, may be you want to clean it up once it's expired and you get the tf diff.

Reply:
Looking at the Rancher API, 'enabled' should just be an attribute not an option. It is functionally useless in Rancher. A user can not disable the resource in the UI or API.

Expired should be an attributed as well.

As an option, we should allow `renew_expired` (bool) so that a user could create a single use token and not renew it after every run if they want to. So if set to false, then an expired token should not show a need to refresh on plan/apply.

Reply:
PR updated, added `renew` argument (default `true`) and set `expired` and `enabled` as attributes…